### PR TITLE
JAMES-2866 MockSMTP server behaviors setup over HTTP

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

File: server/mailet/mock-smtp-server/src/test/java/org/apache/james/mock/smtp/server/MockSMTPBehaviorTest.java [hunk 1]
```diff
@@ -94,4 +125,41 @@ void constructorShouldThrowWhenNumberOfAnswersIsNull() {
...
+            assertThat(behavior)
+                .isEqualTo(POJO_COMPULSORY_FIELDS);
+        }
+
+        @Test
+        void jacksonShouldSerializeMockSMTPBehaviorWithCompulsoryField() throws Exception {
+            String json = OBJECT_MAPPER.writeValueAsString(POJO_COMPULSORY_FIELDS);
+
+            assertThatJson(json)
+                .withOptions(new Options(Option.TREATING_NULL_AS_ABSENT))
```
Comment (Arsnael) [inline server/mailet/mock-smtp-server/src/test/java/org/apache/james/mock/smtp/server/MockSMTPBehaviorTest.java]: Not 100% sure but there is a way I think to set up that directly with the ObjectMapper setup?

Comment (chibenwa) [reply server/mailet/mock-smtp-server/src/test/java/org/apache/james/mock/smtp/server/MockSMTPBehaviorTest.java]: that's some assertThatJson setup not some Jackson setup

File: server/mailet/mock-smtp-server/pom.xml [hunk 2]
```diff
@@ -41,6 +41,10 @@
             <groupId>${james.groupId}</groupId>
             <artifactId>james-core</artifactId>
         </dependency>
+        <dependency>
+            <groupId>${james.groupId}</groupId>
+            <artifactId>james-server-spark</artifactId>
```
Comment (mbaechler) [inline server/mailet/mock-smtp-server/pom.xml]: I don't really like that we expand the use of spark-java in the codebase, it doesn't bring us enough value in comparison of doing things by hand.
Could we use JettyHttpServer instead?

Comment (chibenwa) [reply server/mailet/mock-smtp-server/pom.xml]: Funny, I proposed during the grooming to rely on Play.

The general consensus against me was "let's choose the fastest implementation possible" & "we already all know spark" & "this is testing code" which basically everyone seemed to agree with.

So I respected the overall consensus and proposed that work...

Comment (chibenwa) [reply server/mailet/mock-smtp-server/pom.xml]: Please refer to https://github.com/linagora/james-project/pull/2625 for a Jetty version of this work!

**Comment by chibenwa** [comment]
> the POJO structures now are bit different with the original one

I'm waiting the merge to rebase if you don't mind...

File: server/mailet/mock-smtp-server/pom.xml (hunk 2 above)
Comment (mbaechler) [reply server/mailet/mock-smtp-server/pom.xml]: I would argue the JettyHttpServer is even simpler. Thank you for handling that.

**Comment by chibenwa** [comment]
See https://github.com/linagora/james-project/pull/2625

**Pull request closed without merging**